src/app/qgsmaptoolchamferfillet.cpp

```diff
@@ -0,0 +1,615 @@
+  const QgsGeometry::ChamferFilletOperationType op = QgsMapToolChamferFillet::settingsOperation->value();
+  const int segments = QgsMapToolChamferFillet::settingsFilletSegment->value();
+
+  if ( op == QgsGeometry::ChamferFilletOperationType::Chamfer )
+  {
+    QgsDebugMsgLevel( QStringLiteral( "will chamfer %1 / %2" ).arg( value1 ).arg( value2 ), 3 );
+    newGeom = mManipulatedGeometryInSourceLayerCrs.chamfer( mVertexIndex, value1, value2 );
+  }
+  else
+  {
+    QgsDebugMsgLevel( QStringLiteral( "will fillet %1 / %2" ).arg( value1 ).arg( segments ), 3 );
+    newGeom = mManipulatedGeometryInSourceLayerCrs.fillet( mVertexIndex, value1, segments );
+  }
+
+  if ( newGeom.isNull() )
+  {
+    deleteRubberBandAndGeometry();
+    deleteUserInputWidget();
+    mSourceLayer = nullptr;
+    mGeometryModified = false;
+    emit messageDiscarded();
+    emit messageEmitted( tr( "Creating chamfer/fillet geometry failed: %1" ).arg( mManipulatedGeometryInSourceLayerCrs.lastError() ), Qgis::MessageLevel::Critical );
+  }
+  else
+  {
+    mGeometryModified = true;
+    mModifiedGeometry = newGeom;
+    mRubberBand->setToGeometry( mModifiedGeometry, mSourceLayer );
+  }
+}
+
+
+// ******************
+// Offset User Widget
+
+QgsChamferFilletUserWidget::QgsChamferFilletUserWidget( QWidget *parent )
+  : QWidget( parent )
+{
+  setupUi( this );
+
+  // fill comboboxes
+  mOperationComboBox->addItem( tr( "Chamfer" ), QVariant::fromValue( QgsGeometry::ChamferFilletOperationType::Chamfer ) );
+  mOperationComboBox->addItem( tr( "Fillet" ), QVariant::fromValue( QgsGeometry::ChamferFilletOperationType::Fillet ) );
+
+  QgsGeometry::ChamferFilletOperationType op = QgsMapToolChamferFillet::settingsOperation->value();
+  mOperationComboBox->setCurrentIndex( mOperationComboBox->findData( QVariant::fromValue( op ) ) );
+
+  auto updateLabels = [this]( const QgsGeometry::ChamferFilletOperationType &op ) {
+    if ( op == QgsGeometry::ChamferFilletOperationType::Chamfer )
+    {
+      mVal1Label->setText( tr( "Distance 1" ) );
+      mValue1SpinBox->setDecimals( 6 );
+      mValue1SpinBox->setClearValue( 0.001 );
+      const double value1 = QgsMapToolChamferFillet::settingsValue1->value();
+      mValue1SpinBox->setValue( value1 );
+
+      mVal2Label->setText( tr( "Distance 2" ) );
+      mValue2SpinBox->setDecimals( 6 );
+      mValue2SpinBox->setClearValue( 0.001 );
+      const double value2 = QgsMapToolChamferFillet::settingsValue2->value();
+      mValue2SpinBox->setValue( value2 );
+
+      mVal2Locker->setEnabled( true );
+    }
+    else
+    {
+      mVal1Label->setText( tr( "Radius" ) );
+      mValue1SpinBox->setDecimals( 6 );
+      mValue1SpinBox->setClearValue( 0.001 );
+      const double value1 = QgsMapToolChamferFillet::settingsValue1->value();
+      mValue1SpinBox->setValue( value1 );
+
+      mVal2Label->setText( tr( "Fillet segments" ) );
+      mValue2SpinBox->setDecimals( 0 );
+      mValue2SpinBox->setClearValue( 6.0 );
```

Comment:
min and max values should match those of the setting that will hold this value. Preferably stored on a constexpr so that they stay matched.